From e31549802738de5d2c354d24c4c9063a5840d01a Mon Sep 17 00:00:00 2001
From: Alde Rojas <hello@alde.dev>
Date: Wed, 27 Sep 2023 02:47:29 -0500
Subject: [PATCH] Yield CPU after processing windows messages.

The processing of windows messages results in a loss of 20-40 fps.  This
produces inaccurate framerate limiting. Moving it up resolves the isuse.
---
 Engine/Inc/UnEngineWin.h | 26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/Engine/Inc/UnEngineWin.h b/Engine/Inc/UnEngineWin.h
index 7c0d180..86306f6 100644
--- a/Engine/Inc/UnEngineWin.h
+++ b/Engine/Inc/UnEngineWin.h
@@ -957,19 +957,6 @@ inline void	CMainLoop::RunLoop(void)
 	}
 	unguard;
 
-	// Enforce optional maximum tick rate.
-	guard(EnforceTickRate);
-	if( !GUseFixedTimeStep && IncludeMessaging )
-	{		
-		FLOAT MaxTickRate = Engine->GetMaxTickRate();
-		if( MaxTickRate>0.0 )
-		{
-			FLOAT Delta = (1.0/MaxTickRate) - (appSeconds()-OldTime);
-			appSleep( Max(0.f,Delta) );
-		}
-	}
-	unguard;
-
 	if( IncludeMessaging ) 
 	{
 		// Handle all incoming messages.
@@ -1022,6 +1009,19 @@ inline void	CMainLoop::RunLoop(void)
 		unguard;
 	}
 
+	// Enforce optional maximum tick rate.
+	guard(EnforceTickRate);
+	if( !GUseFixedTimeStep && IncludeMessaging )
+	{
+		FLOAT MaxTickRate = Engine->GetMaxTickRate();
+		if( MaxTickRate>0.0 )
+		{
+			FLOAT Delta = (1.0/MaxTickRate) - (appSeconds()-OldTime);
+			appSleep( Max(0.f,Delta) );
+		}
+	}
+	unguard;
+
 	GStats.DWORDStats( GEngineStats.STATS_Frame_TotalCycles ) = appCycles() - LastFrameCycles;
 	GStats.DWORDStats( GEngineStats.STATS_Game_ScriptCycles ) = GScriptCycles;
 	GScriptCycles	= 0;
-- 
2.31.1.windows.1

